doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+ * Alternatively VRRP can also be configured using native RIF objects without
+ * using VIRTUAL attribute, with the expectation that SAI adapter will consume
+ * resources that will not be used.
+ *
+ * @type bool
+ * @flags CREATE_ONLY
+ * @default false
+ */
+
+SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL,
+
+```
+
+### Configuration and management 
+
+#### Manifest
+
+N/A
+
+#### CLI/YANG Model Enhancements
+
+##### SNOiC VRRP YANG MODEL
+
+```yang
+module sonic-vrrp {
+
+    namespace "http://github.com/Azure/sonic-vrrp";
+    prefix vrrp;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description "VRRP yang Module for SONiC OS";
+
+    revision 2023-08-10 {
+        description "add sonic-vrrp.yang";
+    }
+
+    container sonic-vrrp {
+
+        container VRRP {
+
+            description "VRRP part of configdb.json"
+
+            list VRRP_LIST {
+
+                key "interface_name vrid";
+
+                leaf interface_name {
+                    type union {
+                        type string {
+                            pattern "Ethernet([1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[0-9])";
+                        }
+                        type string {
+                            pattern "PortChannel[0-9]{1,4}";
+                        }
+                        type string {
+                            pattern "Vlan(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[1-9])";
+                        }
+                    } 
+                }
+
+                leaf vrid {
+                    type uint8 {
+                        range 1..255;
+                    }
+                }
+
+                leaf-list vip {
+                    type union {
+                        type inet:ipv4-prefix;
+                        type inet:ipv6-prefix;
+                    }
+                }
+
+                leaf priority {
+                    mandatory true;
+                    type uint8 {
+                        range 1..254;
+                    }
+                }
+
+                leaf adv_interval {
+                    type uint16 {
+                        range 10..40950;
+                    }
+                    default 1000;
+                }
+
+                leaf state {
+                    type enumeration {
+                        enum up;
+                        enum down;
+                    }
+                }
+
+                leaf version {
+                    type enumeration {
+                        enum 2;
+                        enum 3;
+                    }
+                    default 3；
+                }
+
+                leaf preempt {
+                    type enumeration {
+                        enum enabled;
+                        enum disabled;
+                    }
+                }
+
+                leaf-list track_interface {
```

Comment:
Can we have the uplink interface tracking feature as a standalone one instead of combining this with VRRP feature?

Reply:
yes,  will consider to exact it out for a new feature "monitor link" in phase 2.